File: 5_semestre/Programacao-para-Dispositivos-Moveis/UAS/UA1/csharp/Empresa/FolhaPagamento.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Empresa;

public class FolhaPagamento
{
    private readonly Dictionary<string, double> _pagamentosRealizados = new();

    public void ProcessarPagamento(Funcionario f)
    {
        if (!f.Ativo)
        {
            throw new FuncionarioInativoException($"Funcionário {f.Nome} está inativo.");
        }

        double valor = f.CalcularSalarioTotal();
        _pagamentosRealizados[f.Id] = valor;
        Console.WriteLine($"Pagamento de R$ {valor:F2} para {f.Nome}");
    }

    public double TotalPago()       => _pagamentosRealizados.Values.Sum();
    public void LimparRegistros()   => _pagamentosRealizados.Clear();
}

File: 5_semestre/Programacao-para-Dispositivos-Moveis/UAS/UA1/dart/empresa/folha_pagamento.dart
class FolhaPagamento {
  Map<String, double> pagamentosRealizados = {};

  void processarPagamento(Funcionario f) {
    if (!f.ativo) {
      throw FuncionarioInativoException("Funcionário ${f.nome} está inativo.");
    }
    double valor = f.calcularSalarioTotal();
    pagamentosRealizados[f.id] = valor;
    print("Pagamento de R\$ ${valor.toStringAsFixed(2)} para ${f.nome}");
  }

  double totalPago() {
    return pagamentosRealizados.values.fold(0.0, (sum, v) => sum + v);
  }

  void limparRegistros() {
    pagamentosRealizados.clear();
  }
}